tests/integration/metrics_tests/test_metrics.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import time
+import yaml
+
+import ops
+import pytest
+import urllib3
+from pytest_operator.plugin import OpsTest
+from pathlib import Path
+from ..ha_tests import helpers as ha_helpers
+
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+DATABASE_APP_NAME = "mongodb-k8s"
+MONGODB_EXPORTER_PORT = 9216
+MEDIAN_REELECTION_TIME = 12
+
+
+@pytest.fixture(scope="module")
+def chaos_mesh(ops_test: OpsTest) -> None:
+    ha_helpers.deploy_chaos_mesh(ops_test.model.info.name)
+
+    yield
+
+    ha_helpers.destroy_chaos_mesh(ops_test.model.info.name)
+
+
+async def get_address(ops_test: OpsTest, app_name=DATABASE_APP_NAME, unit_num=0) -> str:
+    """Get the address for a unit."""
+    status = await ops_test.model.get_status()  # noqa: F821
+    address = status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+    return address
+
+
+async def verify_endpoints(ops_test: OpsTest, unit: ops.model.Unit) -> str:
+    """Verifies mongodb endpoint is functional on a given unit."""
+    http = urllib3.PoolManager()
+
+    unit_address = await get_address(ops_test=ops_test)
+    mongodb_exporter_url = f"http://{unit_address}:{MONGODB_EXPORTER_PORT}/metrics"
+    mongo_resp = http.request("GET", mongodb_exporter_url)
+
+    assert mongo_resp.status == 200
+
+    # if configured correctly there should be more than one mongodb metric present
+    mongodb_metrics = mongo_resp._body.decode("utf8")
+    assert mongodb_metrics.count("mongo") > 10
```

Comment:
Just curious why we don't use `assertGreater` from `unittest` here

Reply:
If I had to guess I would say because its an unnecessary import

Reply:
I think it is better to use `unitests` library when it is possible